benchmarks/nemo/nemolite2d/manual_versions/psykal_cpp/time_step_mod.f90

```diff
@@ -0,0 +1,94 @@
+module time_step_mod
+    use field_mod
+    use kind_params_mod
+    implicit none
+    public invoke_time_step
+
+    ! Fortran to C wrapper interface using iso_c_bindings.
+    interface
+        subroutine wrapper_c_invoke_time_step( &
+            ! Fields
+            ssha_t, sshn_t, sshn_u, sshn_v, hu, hv, un, vn, ua, ht, ssha_u, &
+            va, ssha_v, &
+            ! Grid
+            tmask, area_t, area_u, area_v, dx_u, dx_v, dx_t, dy_u, dy_v, &
+            dy_t, gphiu, &
+            ! Scalars
+            istp, internal_xstart, internal_xstop, internal_ystart, &
+            internal_ystop, width, rdt, cbfr, visc, omega, d2r, g &
+        ) bind (C, name="c_invoke_time_step")
+            use iso_c_binding
+            real(kind=c_double), intent(inout), dimension(*) :: ssha_t, &
+                sshn_t, sshn_u, sshn_v, hu, hv, un, vn, ua, ht, ssha_u, va, &
+                ssha_v, area_t, area_u, area_v, dx_u, dx_v, dx_t, dy_u, dy_v, &
+                dy_t, gphiu
+            integer(kind=c_int), intent(inout), dimension(*) :: tmask
+            integer(kind=c_int), intent(in), value :: istp, internal_xstart, &
+                internal_xstop, internal_ystart, internal_ystop, width
+            real(kind=c_double), intent(in), value :: rdt, cbfr, visc, omega, &
+                d2r, g
+        end subroutine wrapper_c_invoke_time_step
+    end interface    
+
+contains
+
+    ! This invoke_time_step needs to fetch all necessary arrays, global
+    ! variables, and the additional scalar values needed during the PSy-layer
+    ! implementation, and pass them all to the wrapper interface.
+    SUBROUTINE invoke_time_step(istp, ssha_t, ssha_u, ssha_v, sshn_t, sshn_u, &
+        sshn_v, hu, hv, ht, ua, va, un, vn)
+
+        USE physical_params_mod, ONLY: omega, d2r, g
+        USE model_mod, ONLY: rdt, cbfr, visc
+
+        implicit none
+
+        TYPE(r2d_field), intent(inout) :: ssha_t, sshn_t, sshn_u, &
+            sshn_v, hu, hv, un, vn, ua, ht, ssha_u, va, ssha_v
+        INTEGER, intent(in) :: istp
+
+        ! TODO: Should this use %get_data() instead?
```

Comment:
The compiler didn't like it. It says the get_data() can not be passed to an INOUT parameter. It probably can be solved with a temporary reference that then is passed to c wrapper, but since the %data already works I would prefer to think more carefully about this in a future PR.

Reply:
Fine. Thanks for trying.

Reply:
Please could you create an issue for this and reference it here.